### PR TITLE
BUG: 256 colors of tiff palette were read, only correct for 8bit images

### PR DESCRIPTION
**Pull request by YannLePoul**
## PR Checklist

- [] :no_entry_sign: [Makes breaking changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes)
- [] :no_entry_sign: [Makes design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes)
- [] :no_entry_sign: Adds the License notice to new files.
- [] :no_entry_sign: Adds Python wrapping.
- [] :no_entry_sign: Adds tests and baseline comparison (quantitative).
- [] :no_entry_sign: [Adds test data](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Documentation/UploadBinaryData.md).
- [] :no_entry_sign: Adds Examples to [ITKExamples](https://github.com/InsightSoftwareConsortium/ITKExamples)
- [] :no_entry_sign: Adds Documentation.

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by dzenanz** [comment]
Does this supersede #1218?

**Comment by YannLePoul** [comment]
It is two independent bugs.

> Does this supersede #1218?

It is two independent bugs, that's why I made two pull request because. The two pull request are rebasing well on top of each other. But I could merge them if you think it is more convinient

**Comment by dzenanz** [comment]
The warning should be easy enough to fix.

File: Modules/IO/TIFF/src/itkTIFFImageIO.cxx [hunk 1]
```diff
@@ -960,17 +960,27 @@ TIFFImageIO::ReadRawByteFromTag(unsigned int t, unsigned int & value_count)
...
-  m_ColorPalette.resize(256);
-  for (unsigned int cc = 0; cc < 256; ++cc)
+  if (m_TotalColors > 0)
   {
-    unsigned short red, green, blue;
-    this->GetColor(cc, &red, &green, &blue);
+    // if a palette have been found store it in m_ColorPalette
+    unsigned int nColors = static_cast<unsigned int>(m_TotalColors);
+    m_ColorPalette.resize(nColors);
+    for (unsigned int cc = 0; cc < static_cast<unsigned int>(nColors); ++cc)
```
Comment (dzenanz) [inline Modules/IO/TIFF/src/itkTIFFImageIO.cxx]: Since `nColors` is already unsigned, this `static_cast` is not needed.

Comment (YannLePoul) [reply Modules/IO/TIFF/src/itkTIFFImageIO.cxx]: oups!

Comment (dzenanz) [reply Modules/IO/TIFF/src/itkTIFFImageIO.cxx]: Your latest push c4d417f6f69e9b9f2b7209adf0c6ff0d3c7e8d49 didn't seem to have changed anything.

Comment (YannLePoul) [reply Modules/IO/TIFF/src/itkTIFFImageIO.cxx]: reoups...

**Comment by YannLePoul** [comment]
Is python build test fail related ? Doesn't seems so

**Pull request merged**